pyart/graph/radarmapdisplay_cartopy.py

```diff
@@ -0,0 +1,510 @@
+                                            lon + lon_offset,
+                                            lat + lat_offset,
+                                            src_crs=cartopy.crs.PlateCarree())
+            self.basemap.annotate(label_text, xy=(x_text, y_text))
+
+    def plot_line_geo(self, line_lons, line_lats, line_style='r-', **kwargs):
+        """
+        Plot a line segments on the current map given values in lat and lon.
+
+        Additional arguments are passed to basemap.plot.
+
+        Parameters
+        ----------
+        line_lons : array
+            Longitude of line segment to plot.
+        line_lats : array
+            Latitude of line segment to plot.
+        line_style : str
+            Matplotlib compatible string which specifies the line style.
+
+        """
+        self._check_basemap()
+        self.basemap.plot(
+            line_lons, line_lats, line_style,
+            transform=cartopy.crs.PlateCarree(), **kwargs)
+
+    def plot_line_xy(self, line_x, line_y, line_style='r-', **kwargs):
+        """
+        Plot a line segments on the current map given radar x, y values.
+
+        Additional arguments are passed to basemap.plot.
+
+        Parameters
+        ----------
+        line_x : array
+            X location of points to plot in meters from the radar.
+        line_y : array
+            Y location of points to plot in meters from the radar.
+        line_style : str, optional
+            Matplotlib compatible string which specifies the line style.
+
+        """
+        self._check_basemap()
+        self.basemap.plot(line_x, line_y, line_style, transform=self.proj,
+                          **kwargs)
+
+    def plot_range_ring(self, range_ring_location_km, npts=360,
+                        line_style='k-', **kwargs):
+        """
+        Plot a single range ring on the map.
+
+        Additional arguments are passed to basemap.plot.
+
+        Parameters
+        ----------
+        range_ring_location_km : float
+            Location of range ring in km.
+        npts: int
+            Number of points in the ring, higher for better resolution.
+        line_style : str
+            Matplotlib compatible string which specified the line
+            style of the ring.
+
+        """
+        # The RadarDisplay.plot_range_rings uses a col parameter to specify
+        # the line color, deal with this here.
+        if 'col' in kwargs:
+            color = kwargs.pop('col')
+            kwargs['c'] = color
+        if 'ax' in kwargs:
+            kwargs.pop('ax')
+        self._check_basemap()
+        angle = np.linspace(0., 2.0 * np.pi, npts)
+        xpts = range_ring_location_km * 1000. * np.sin(angle)
+        ypts = range_ring_location_km * 1000. * np.cos(angle)
+        self.plot_line_xy(xpts, ypts, line_style=line_style, **kwargs)
+
+# Thses methods are a hack to allow gridlines when the projection is lambert
+# http://nbviewer.jupyter.org/gist/ajdawson/dd536f786741e987ae4e
```

Comment:
Do we have permission from the original author to include these?  Would they allow this work to be BSD licensed?